lib/src/test/java/io/ably/lib/test/realtime/RealtimePresenceTest.java

```diff
@@ -1979,38 +1988,64 @@ public void onSuccess() {
 
 				@Override
 				public void onError(ErrorInfo reason) {
-					fail("Presence: enter() failed");
 				}
 			});
-
-			try {
-				synchronized (presenceWaiter) {
-					while (!presenceWaiter[0])
+			synchronized (presenceWaiter) {
+				while (!presenceWaiter[0]) {
+					try {
 						presenceWaiter.wait();
+					} catch (InterruptedException e) {}
 				}
 			}
-			catch (InterruptedException e) {
-			}
+
+			/*
+			 * We put testClientId2 presence data into the client library presence map but we
+			 * don't send it to the server
+			 */
+
+			MockWebsocketFactory.blockSend();
+			channel.presence.enterClient(testClientId2, presenceData);
+
+			ProtocolMessage msg = new ProtocolMessage();
+			msg.connectionId = connId;
+			msg.action = ProtocolMessage.Action.sync;
+			msg.channel = channelName;
+			msg.presence = new PresenceMessage[] {
+					new PresenceMessage() {{
+						action = Action.present;
+						id = String.format("%s:0:0", connId);
+						timestamp = System.currentTimeMillis();
+						clientId = testClientId2;
+						connectionId = connId;
+						data = presenceData;
+					}}
+			};
+			ably.connection.connectionManager.onMessage(msg);
+
+			MockWebsocketFactory.allowSend();
 
 			ably.connection.connectionManager.requestState(ConnectionState.suspended);
 			channelWaiter.waitFor(ChannelState.suspended);
 
+			/*
+			 * When restoring from suspended state server will send sync message erasing
+			 * testClientId2 record from the presence map. Client should re-send presence message
+			 * for testClientId2 and restore its presence data.
+			 */
+
 			ably.connection.connectionManager.requestState(ConnectionState.connected);
 			channelWaiter.waitFor(ChannelState.attached);
 
 			try {
 				Thread.sleep(500);
-				PresenceMessage[] presenceMessages = channel.presence.get(testClientId1, true);
-				assertNotNull("Verify there are presence messages after reattach");
-				assertEquals("Verify there is exactly one presence message", presenceMessages.length, 1);
-				assertEquals("Verify presence message data is the same as before", presenceMessages[0].data, presenceData);
-			}
-			catch (InterruptedException e) {
-			}
+				assertEquals("Verify correct presence message data has been received",
+						channel.presence.get(testClientId2, true)[0].data, presenceData);
```

Comment:
I still don't think that this does what you think. Calling (realtime)Channel.presence.get()` gets the value from the local `PresenceMap`; so by this test you still don't know that the update has reached the server as a result of reattaching after `suspended`.

If this was a REST `Channel.presence.get()`, you'd be certain that you're getting the value from the server.